### PR TITLE
Create the associated bugs found and configuration mode

### PR DESCRIPTION
**Pull request by boehlergerman**
Issues #99 
- [
- Notification of errors when processing the 10 most used javascript libraries
- Test structure or dependency injector
- configuration mode for dependency injector and future configurations
 ] 

- tests on phantomjs are pending to work with external cdn Issues #109

### CODE REVIEW

File: deps-injector/deps-injector.js [hunk 1]
```diff
@@ -6,57 +6,85 @@ require.config({
...
     };
 
-    PumaInjector.prototype.installDeps = function(deps){
+    PumaInjector.prototype.installDeps = function (deps) {
         var dependencyList = deps || this.deps;
 
         //Retrieve the script from CDN's
-        for(var i=0;i<dependencyList.length;i++){
-            this.getExternalScript(dependencyList[i]);
+        for (var key in dependencyList) {
```
Comment (emravera) [inline deps-injector/deps-injector.js]: Style C for is more performant in JS.

File: src/runtime.js [hunk 2]
```diff
@@ -54,13 +54,26 @@ define([
     /**
      * @constructor
      */
+    function RuntimeConfig() {
+        this._modeConfig = 'default';
```
Comment (emravera) [inline src/runtime.js]: For states you can use constants like:

```
RuntimeConfig.MODES = {
       DEFAULT: 'default',
       TEST: 'test'
};
```

And you use to compare against these enumeration to make more maintainable

File: deps-injector/deps-injector.js (hunk 1 above)
Comment (boehlergerman) [reply deps-injector/deps-injector.js]: Well, I make puma.js an array and not an object.

File: src/runtime.js (hunk 2 above)
Comment (boehlergerman) [reply src/runtime.js]: I agree

**Pull request merged**